File: 코드/ch3/clip5/boj15657/main.py
def print_seq(arr):
    print(" ".join(map(str, arr)))

def perm(arr, output, depth, n, m, start):
    if depth == m:
        print_seq(output)
        return

    for i in range(start, n):
        output[depth] = arr[i]
        perm(arr, output, depth + 1, n, m, i)

n, m = map(int, input().split())
numbers = list(map(int, input().split()))
numbers.sort()

perm(numbers, [0] * m, 0, n, m, 0)

File: 코드/ch3/clip5/boj15657/Main.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static StringBuilder sb = new StringBuilder();

    public static void print(int[] arr, int n) {
        for (int i = 0; i < n; i++) {
            sb.append(arr[i]).append(" ");
        }
        sb.append("\n");
    }

    public static void perm(int[] arr, int[] output, int depth, int n, int m, int start) {
        if (depth == m) {
            print(output, m);
            return;
        }

        for (int i = start; i < n; i++) {
            output[depth] = arr[i];
            perm(arr, output, depth + 1, n, m, i);
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();

        int[] numbers = new int[n];
        for (int i = 0; i < n; i++) {
            numbers[i] = sc.nextInt();
        }
        Arrays.sort(numbers);
        perm(numbers, new int[m], 0, n, m, 0);
        System.out.println(sb);
    }
}
